### ISSUE TITLE
Add a Color property for Badge component

### ISSUE THREAD

**Issue by FredericHertz** [Sage/carbon#5839]
### Desired behaviour

I would like the Badge component to have the same color as its children by adding a Color property.

### Current behaviour

The Badge component always have 'primary' color.

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by DipperTheDan** [comment]
@ljemmo @clairedenning
Would you be able to review this enhancement request please and thank you 😄

**Comment by clairedenning** [comment]
What children can a badge have?

**Comment by FredericHertz** [comment]
Box or Button component

**Comment by nineteen88** [comment]
@ljemmo @clairedenning could you update on this please?

**Comment by ljemmo** [comment]
@nineteen88 or @FredericHertz please can you DM claire and I with the design proposal so we can review.

**Comment by ljemmo** [comment]
@edleeks87 i think i'd need to see what he's trying to achieve via a design to fully understand this requirement. Ultimately we try and keep components locked down to drive consistency where we can. Unless there's a good enough reason to do this, i'd suggest we keep it simple and not include this prop (happy to be proved wrong though).

**Comment by ljemmo** [comment]
@sianford i've not seen anything just yet.

**Comment by ljemmo** [comment]
@Parsium I just caught up with Frederic and it looks they rightly want to be able to override color of badge to red when placed over a destructive button. I note also that harpal wants to change badge to blue within the global nav, so multiple designers want this. I suggest we allow an override prop that allows them to inject whatever color token they require. Sound ok?

**Comment by ljemmo** [comment]
Screenshot of the problem for further context (no commercially sensitive info) ![image](https://github.com/Sage/carbon/assets/98600534/268c8de1-beeb-4cb3-9b5a-7adb622f640d)

**Comment by Parsium** [comment]
Brilliant, thanks for this @ljemmo 👍🏼 Makes sense, let's get this on our backlog then

**Comment by Parsium** [comment]
@ljemmo would Design System be able to update the docs to mention about the ability to add any colour? Just so we are making sure that DS is the source of truth when it comes to styling decisions

**Comment by ljemmo** [comment]
@Parsium i'll add a ticket to our backlog. FYI, placement seems to be an issue of the badge on top of our new rounded buttons. Can we fix that alongside your ticket?

**Comment by Parsium** [comment]
Thanks @ljemmo 👍🏼 Personally I would suggest amending the placement in DS designs first, so we have some concrete designs to work from when we pick up the ticket to implement this in Carbon

**Comment by ljemmo** [comment]
@Parsium  that work has already been done - just the color element needs updating in our guidance. See below for how we'd like the badge to be placed. This is updated in our component figma file too. 
![Screenshot 2023-05-17 at 11 02 47](https://github.com/Sage/carbon/assets/98600534/f1688a0b-5195-41db-a3cc-a889069411a9)

**Comment by Parsium** [comment]
Ah brilliant cheers @ljemmo, I would raising it as a separate ticket, but both of these tasks could be done in parallel

FE-5894
